src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs

```diff
@@ -72,6 +83,33 @@ public static void Register(HttpConfiguration config, WebHostSettings settings =
             config.InitializeReceiveSalesforceWebHooks();
         }
 
+        private static void PrependFoldersToEnvironmentPath()
+        {
+            string home = Environment.GetEnvironmentVariable("HOME");
+
+            // Only do this when %HOME% is defined (normally on Azure)
+            if (!String.IsNullOrEmpty("HOME"))
+            {
+                // Create the tools folder if it doesn't exist
+                string toolsPath = Path.Combine(home, @"site\tools");
+                Directory.CreateDirectory(toolsPath);
+
+                var folders = new List<string>();
+                folders.Add(Path.Combine(home, @"site\tools"));
+
+                string path = Environment.GetEnvironmentVariable("PATH");
+                string additionalPaths = String.Join(";", folders);
+
+                // Make sure we haven't already added them. This can happen if the appdomain restart (since it's still same process)
```

Comment:
Could you introduce a static intitialize flag that you use to ensure you only do this init once and only once?